### ISSUE TITLE
Compare different gradient descent algorithms & regressions for linear regression

### ISSUE THREAD

**Issue by anton-dergunov** [anton-dergunov/ml-playground#7]
Compare:
- Batch GD
- Stochastic GD
- Mini-batch GD
and:
- Ridge
- Lasso
- Elastic Net

Broader: compare different GD algorithms, visualize like on mlpack page:
http://www.mlpack.org/docs/mlpack-git/doxygen/optimizertutorial.html
https://machinelearningmastery.com/adam-optimization-algorithm-for-deep-learning/
https://arxiv.org/abs/1609.04747
http://ruder.io/optimizing-gradient-descent/index.html
http://ruder.io/optimizing-gradient-descent/index.html
https://www.tensorflow.org/versions/r0.12/api_docs/python/train/optimizers
